pkg/bootstrap/platform/gcp.go

```diff
@@ -211,3 +228,51 @@ func (e *gcpEnv) Locality() *core.Locality {
 
 	return &l
 }
+
+// Labels attempts to retrieve the GCE instance labels from provided metadata within the timeout
```

Comment:
Let's add a small comment here about needing read-level access to the compute API.